docs/adr/0024-upgrade-templating-toolset-of-auth-server-emails.md

```diff
@@ -0,0 +1,177 @@
+# Upgrade the templating toolset of Auth Server emails
+
+- Status: accepted
+- Deciders: Yogita Bhatia, Lauren Zugai, Jody Heavener, Vijay Budhram, Ben Bangert, Les Orchard, Danny Coates
+- Date: 2021-06-17
+
+Technical Story: [#4627](https://github.com/mozilla/fxa/issues/4627) / [#8600](https://github.com/mozilla/fxa/issues/8600)
+
+## Context and Problem Statement
+
+[Mustache](https://www.npmjs.com/package/mustache), used in `fxa-auth-server` email templates, can accommodate very limited logic in its templates which has been a pain point for Firefox Accounts engineers and discussion around using a different templating system [began here](https://github.com/mozilla/fxa/issues/4627). While converting our emails to a more modern templating solution, there is an opportunity to evaluate what stack would be the most ideal for FxA emails beyond a proposed templating solution. This includes evaluating our CSS options, improving how we can preview emails in various states, and our localization tools, and the best approach for landing new templates in production.
+
+## Decisions and Considered Options
+
+### Templating and Styling
+
+- Option A - Continue to use Mustache and ad-hoc inline styles
+- Option B - Use React server-side to generate static HTML templates with TailwindCSS
+- Option C - Use EJS and MJML, using CSS options offered by MJML
+
+### Email previewing
+
+- Option A - Continue to use the `write-emails` command
+- Option B - Use Storybook
+- Option C - Use Mailtrap
+
+--
+
+Furthermore, there are a few **other decisions** worth noting that won’t necessarily have pros/cons lists:
+
+- How to handle generating plaintext versions
+- Transitioning to Fluent over GetText for localization
+- Plans around integration involving feature flagging and the QA process
+
+## Decision Outcomes
+
+### Templating and Styling
+
+Chosen option: "Option C - Use EJS and MJML, using CSS options offered by MJML", because:
+
+- HTML email has a lot of quirks - MJML shifts the burden of maintaining solutions for these off of FxA engineers now and in the future
+- While we use React and Tailwind in other parts of FxA, React is heavy for an email solution since no state is involved, component reuse across FxA would likely be very minimal, and it involves a more complex build setup than EJS
+- MJML helps significantly with responsiveness in emails, reducing time spent developing templates and making future email redesigns easier
+
+### Email Previewing
+
+Chosen option: "Option B - Use Storybook", because:
+
+- It is easy to setup and provides us the flexibility to preview all of the different email states together.
```

Comment:
Ohh, we should take out "it is easy to setup". That was the case when we picked the React option.

Also tiny nit, some of these have periods (as well as other sections) when other points don't